src/System.Private.ServiceModel/src/System/ServiceModel/Channels/SocketConnection.cs

```diff
@@ -92,13 +92,27 @@ protected string RemoteEndpointAddressString
 
         protected static void OnReceiveTimeout(SocketConnection socketConnection)
         {
-            socketConnection.Abort(SR.Format(SR.SocketAbortedReceiveTimedOut, socketConnection._receiveTimeout), TransferOperation.Read);
+            try
+            {
+                socketConnection.Abort(SR.Format(SR.SocketAbortedReceiveTimedOut, socketConnection._receiveTimeout), TransferOperation.Read);
+            }
+            catch (SocketException)
+            {
+                // Guard against unhandled SocketException in timer callbacks
```

Comment:
any idea on what to do with this exception? 
The connection is gone so any attempt to use it will result in exceptions on the calling thread. There is a way to save the exception to (presumably) throw it later but this will eventually result in a much bigger change in the behavior.